Src/WitsmlExplorer.Frontend/services/credentialsService.ts

```diff
@@ -78,76 +83,69 @@ class CredentialsService {
-  }
-
-  public hasPasswordForUrl(serverUrl: string) {
-    return this.credentials.find((c) => c.server.url === serverUrl)?.password !== undefined;
+    return this.hasValidCookieForServer(server?.url);
   }
 
   public hasValidCookieForServer(serverUrl: string): boolean {
     //use local storage to check whether the cookie is valid, because the cookie is httpOnly
-    const cookieTimestamp = localStorage.getItem(serverUrl);
-    if (!cookieTimestamp) {
+    const cookieInfo = localStorage.getItem(serverUrl);
+    if (!cookieInfo) {
       return false;
+    } else {
+      const cInfo: CookieInfo = JSON.parse(cookieInfo);
+      return new Date().getTime() < new Date(cInfo.expiry).getTime();
     }
-    return new Date().getTime() < new Date(cookieTimestamp).getTime();
-  }
-
-  public keepLoggedInToServer(serverUrl: string): boolean {
-    return !!localStorage.getItem(serverUrl);
   }
 
-  public async verifyCredentials(credentials: BasicServerCredentials, abortSignal?: AbortSignal): Promise<any> {
-    const response = await ApiClient.get(`/api/credentials/authorize`, abortSignal, [credentials]);
-    if (response.ok) {
-      return response.json();
-    } else {
-      const { message }: ErrorDetails = await response.json();
-      CredentialsService.throwError(response.status, message);
+  private isJsonString(str: string) {
+    try {
+      JSON.parse(str);
+    } catch (e) {
+      return false;
     }
+    return true;
   }
 
-  public async verifyCredentialsWithCookie(credentials: BasicServerCredentials, abortSignal?: AbortSignal): Promise<BasicServerCredentials> {
-    const response = await ApiClient.get(`/api/credentials/authorizewithcookie`, abortSignal, [credentials], true);
-    if (response.ok) {
-      const cookie = await response.json();
-      const decoded = Buffer.from(cookie, "base64").toString();
-      const creds = decoded.split(":");
-      return {
-        server: credentials.server,
-        username: creds[0],
-        password: creds[1]
-      };
-    } else {
-      const { message }: ErrorDetails = await response.json();
-      CredentialsService.throwError(response.status, message);
+  private allLocalStorage() {
+    const values = [];
+    const keys = Object.keys(localStorage);
+    let i = keys.length;
+    while (i--) {
+      if (keys[i].startsWith("http") && this.isJsonString(localStorage.getItem(keys[i]))) {
+        values.push(localStorage.getItem(keys[i]));
+      }
     }
+    return values;
+  }
+
+  // refresh timestamp for all "session" entries in localstorage, to match session cookies
+  public refreshLocalstorageSessions() {
+    const expirationTime = new Date();
+    expirationTime.setHours(expirationTime.getHours() + 1);
+    const cookies: CookieInfo[] = this.allLocalStorage()
+      .map((n) => JSON.parse(n))
+      .filter((n) => n.type == "WExSession");
+    cookies.forEach((element) => {
+      localStorage.removeItem(element.url);
```

Comment:
Unnecessary removeItem call

Reply:
I tried to only use `setItem` but the item did not update unless I removed it first.

Reply:
Hmm, on my end the item updates without removing it first.

Reply:
I'll leave it there for now, it might be different behaviours per browser, but we should remove it if it behaves consistently